concordium-consensus/src/Concordium/GlobalState/Persistent/LMDB.hs

```diff
@@ -688,3 +693,44 @@ writeFinalizationComposite finRec blocktss = resizeOnFull (finRecSize + blocksSi
           digestSize + 64 + digestSize + (32 * Prelude.length vs) + 48 + 64
     blocksSize = sum . map (\b -> 2*digestSize + fromIntegral (LBS.length . fst . fst $ b)) $ serializedBlocksTss
     tssSize = (* (2 * digestSize + 16)) . sum . map (Prelude.length . snd) $ blocktss
+
+-- If an explicitly finalized block does not satisfy a predicate, delete it, together with its
+-- finalization record, all blocks implicitly finalized by its finalization records and all
+-- associated transaction statuses. Returns the last finalized block hash available in the unrolled
+-- treestate/blockstate on success.
+unrollTreeStateWhile :: (MonadLogger m, MonadIO m, IsProtocolVersion pv, FixedSizeSerialization st) =>
```

Comment:
Where do you use the MonadLogger here? I can't see any logging.

Reply:
I was using it during debugging, and I think I'll still keep using it to log the deleted blocks and finalization records without having to return them to `loadSkovPersistentState`.

Reply:
Sounds reasonable.